File: apps/api/app/schemas/asset.py
"""Scene asset schemas."""

import uuid

from pydantic import BaseModel, Field


class SceneAssetPublic(BaseModel):
    id: uuid.UUID
    scene_id: uuid.UUID
    asset_type: str | None = None
    description: str | None = None
    image_url: str | None = None
    status: str

    model_config = {"from_attributes": True}


class SceneAssetGroup(BaseModel):
    """Assets grouped under their scene, for the review screen."""

    scene_number: int
    scene_label: str | None = None
    assets: list[SceneAssetPublic]


class AssetPlanState(BaseModel):
    project_status: str
    pending_assets: int
    scenes: list[SceneAssetGroup]


class AssetCostPreview(BaseModel):
    pending_assets: int
    credits_required: float
    estimated_usd: float
    current_balance: float
    sufficient: bool


class AssetStatusUpdate(BaseModel):
    status: str = Field(pattern="^(approved|rejected)$")

File: apps/web/types/asset.ts
export type AssetStatus =
  | "pending"
  | "generated"
  | "approved"
  | "rejected"
  | "user_uploaded"
  | "failed";

export interface SceneAssetPublic {
  id: string;
  scene_id: string;
  asset_type: string | null;
  description: string | null;
  image_url: string | null;
  status: AssetStatus | string;
}

export interface SceneAssetGroup {
  scene_number: number;
  scene_label: string | null;
  assets: SceneAssetPublic[];
}

export interface AssetPlanState {
  project_status: string;
  pending_assets: number;
  scenes: SceneAssetGroup[];
}

export interface AssetCostPreview {
  pending_assets: number;
  credits_required: number;
  estimated_usd: number;
  current_balance: number;
  sufficient: boolean;
}
